functions/src/modules/pm/categories/createCategory.ts

```diff
@@ -1,5 +1,55 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import {
+  expressErrorHandlerFactory,
+  requestBodyTypeValidator,
+} from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const createCategory: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: validate },
+    sponsor: { accepted: validate },
+  })(req, res, next);
+};
 
-export default replaceMe;
+const validate: ExpressFunction = (req, res, next) => {
+  const validationRules: ValidatorObjectRules = {
+    type: "object",
+    rules: {
+      categoryId: { rules: ["string"], required: true },
+      prizeCategoryName: { rules: ["string"], required: true },
+      prizeCategoryDescription: { rules: ["string"], required: true},
+      approvalStatus: { rules: [{
+        type: "enum",
+        rules: ["string"],
```

Comment:
The `rules` for an enum are the allowed values; if we keep this as string, the only accepted value would be "string" itself. Therefore, this should be the values itself that we want to allow; likely "approved", "rejected", "inProgress", or "awaitingApproval" for organizers, and just "inProgress" or "awaitingApproval" for sponsors.
```suggestion
        rules: (res.locals.permissions as UserPermission).role === "ORGANIZER ? ["approved", "rejected", "inProgress", "awaitingApproval"] : ["inProgress", "awaitingApproval"],
```